### PR TITLE
Implement `output_size` for `osquery_schedule`

### PR DESCRIPTION
**Pull request by sharvilshah**
Addresses #7147

This PR implements a simple `output_size` column for scheduled queries, while being aware of the newer typed `RowData`. 

This was also discussed in the last office hours.

### CODE REVIEW

**Review by zwass — COMMENTED** [review]
The approach looks good to me.

A couple of requests:

1) Can we add a unit test with a query that always returns a fixed amount of data (eg. `select 1, 2;`)?

2) Can you please update the documentation to indicate what the output size means? My understanding is that we're making a cumulative count of how much output has been generated by the query since osquery started? But I'm not sure whether it's the size of the JSON output or the in memory result set or something else.

**Comment by sharvilshah** [comment]
Thanks for the quick review @zwass

> 1. Can we add a unit test with a query that always returns a fixed amount of data (eg. select 1, 2;)?

Yep, I will work on this. This type of test might ended up being a bit more integration style, actually scheduling a query and looking at the result -- and might require breaking it down by platform. I will look into it more.

> 2. Can you please update the documentation to indicate what the output size means? My understanding is that we're making a cumulative count of how much output has been generated by the query since osquery started? But I'm not sure whether it's the size of the JSON output or the in memory result set or something else.

I will take a closer look and update the docs. From what I gather, before the refactor, `RowData` was just a map `map<std::string, std::string>`, and back then `output_size` was the size of both strings (key and value) for each rown. This is probably why it closely matched the JSON output. 

Now that `RowData` is typed and named `RowDataTyped` which is a map of string to a boost variant: `map<std::string, boost::variant>`, basically a fancy templated union of a few types, `output_size` is now the size of key (string) and `sizeof` whatever the type the value is. 

Long winded way of saying that `output_size` is now the size of the rows (result) in memory.

File: osquery/sql/sqlite_util.cpp [hunk 1]
```diff
@@ -223,6 +223,25 @@ class StringEscaperVisitor : public boost::static_visitor<> {
   }
 };
 
+class SizeVisitor : public boost::static_visitor<> {
+ public:
+  template <typename T>
+  void operator()(T& t) const {
+    size += sizeof(t);
```
Comment (npamnani) [inline osquery/sql/sqlite_util.cpp:230]: size += sizeof(t); or size = sizeof(t);?

If type is string m I believe there should be an overloaded operator for that sth like ?
void operator()(const std::string& t) {
  size = t.size();
}

File: osquery/sql/sqlite_util.cpp [hunk 2]
```diff
@@ -223,6 +223,25 @@ class StringEscaperVisitor : public boost::static_visitor<> {
...
+  uint64_t get_size() const {
+    return size;
+  }
+
+  SizeVisitor() {
+    size = 0;
+  }
+
+ private:
+  mutable uint64_t size;
```
Comment (npamnani) [inline osquery/sql/sqlite_util.cpp:242]: mutable? visitor is not cont type.

Comment (theopolis) [reply osquery/sql/sqlite_util.cpp:242]: perhaps if `operator()` took a const reference?

Comment (npamnani) [reply osquery/sql/sqlite_util.cpp:242]: I believe purpose of mutable specifier is to add an exception, amongst other class members and there is only one member, I dont see any exceptional scenario here.

**Comment by sharvilshah** [comment]
messed up the rebase :/  fixing

**Comment by sharvilshah** [comment]
closing this in favor of #7436

**Pull request closed without merging**